examples/EvaluatingChatAssistant/EvaluatingChatAssistant.ipynb

```diff
@@ -0,0 +1,353 @@
+    "- `input` is the last `user` turn that will be sent to the completion\n",
+    "- `expected` is the output expected given the input\n",
+    "\n",
+    "From looking at this one example, it is clear why we need the history to be able to provide a helpful response.\n",
+    "\n",
+    "If you were asked \"who won the men's trophy that year?\", you would wonder *What trophy? Which year?*. But if you read the `chat_history`, you'd be able to answer the question (maybe after some quick research)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "const dataset = [\n",
+    "  {\n",
+    "    chat_history: [\n",
+    "      { role: \"user\", content: \"when was the ballon d'or first awarded for female players?\" },\n",
+    "      {\n",
+    "        role: \"assistant\",\n",
+    "        content:\n",
+    "          `The Ballon d'Or Féminin, awarded to the best female football player in the world, was first introduced in 2018.\n",
+    "          The inaugural winner was Ada Hegerberg of Norway, who played for Olympique Lyonnais at the time.`,\n",
+    "      },\n",
+    "    ],\n",
+    "    input: \"who won the men's trophy that year?\",\n",
+    "    expected: \"The men's Ballon d'Or trophy in 2018 was won by Luka Modrić of Croatia, who played for Real Madrid.\"\n",
+    "  },\n",
+    "  // ... rest of dataset\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Running evals\n",
+    "\n",
+    "The key to running evals on a multi-turn conversation is to include the history of the chat in the chat completion request.\n",
+    "\n",
+    "To do so, we simply add the history between the `system` message and the final `user` message in the `messages` argument as seen below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import { wrapOpenAI } from \"braintrust\";\n",
+    "import { OpenAI } from \"openai\";\n",
+    "\n",
+    "async function runTask({\n",
+    "  input,\n",
+    "  chat_history\n",
+    "}: {\n",
+    "  input: string,\n",
+    "  chat_history: {\n",
+    "    role: \"user\" | \"assistant\";\n",
+    "    content: string;\n",
+    "  }[]\n",
+    "}) {\n",
+    "  const client = wrapOpenAI(\n",
+    "    new OpenAI({\n",
+    "      baseURL: \"https://braintrustproxy.com/v1\",\n",
+    "      apiKey: process.env.OPENAI_API_KEY ?? \"\", // Can use OpenAI, Anthropic, Mistral etc. API keys here\n",
+    "    })\n",
+    "  )\n",
+    "\n",
+    "  const response = await client.chat.completions.create({\n",
+    "    model: \"gpt-4o\",\n",
+    "    messages: [\n",
+    "      {\n",
+    "        role: \"system\",\n",
+    "        content: \"You are a helpful and polite assistant who knows about sports.\"\n",
+    "      },\n",
+    "      ...chat_history,\n",
```

Comment:
maybe add a comment here indicating that this is where the chat history is added